### PR TITLE
nixos/dendrite: rename http(s)Port to http(s)Address

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jian-lin** [comment]
Thanks for your work.

The change from port to address enables dendrite listen on some specified address like localhost instead of all addresses, which is good when using a reverse proxy.
Also, it is consistent with upstream.

However, IMHO, it has some disadvantages:
1. backward incompatibility
2. nixos user needs to find the format of the new option `http(s)Address`

What about adding a new option like `ip` defaulting to "" (empty string) instead of changing the option `http(s)Port` to `http(s)Address` or at least adding an example for `http(s)Address`?

**Comment by NickCao** [comment]
`http(s)Address` the name itself should be clean enough, and the default value is visible in the generated document as a hint. For backward compatibility, we may keep `http(s)Port` and default `http(s)Address` to that, but that would make the module a lot messier.

File: nixos/doc/manual/from_md/release-notes/rl-2205.section.xml [hunk 1]
```diff
@@ -1279,6 +1279,15 @@
           Plugins are automatically repackaged using autoPatchelf.
         </para>
       </listitem>
+      <listitem>
+        <para>
+          <literal>services.dendrite.httpPort</literal> and
+          <literal>services.dendrite.httpsPort</literal> has been
```
Comment (jian-lin) [inline nixos/doc/manual/from_md/release-notes/rl-2205.section.xml:1285]: has -> have

File: nixos/doc/manual/release-notes/rl-2205.section.md [hunk 2]
```diff
@@ -467,6 +467,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 - `services.mattermost.plugins` has been added to allow the declarative installation of Mattermost plugins.
   Plugins are automatically repackaged using autoPatchelf.
 
+- `services.dendrite.httpPort` and `services.dendrite.httpsPort` has been replaced with `services.dendrite.httpAddress` and `services.dendrite.httpsAddress` respectively.
```
Comment (jian-lin) [inline nixos/doc/manual/release-notes/rl-2205.section.md:470]: has -> have

**Review by jian-lin — APPROVED** [review]
Looks good to me except the syntax error in the release note.

**Pull request closed without merging**